tests/helper.py

```diff
@@ -2,6 +2,195 @@
+        for i in range(n_samples):
+            expected[i] = self.metric.exp(one_tangent_vec[i], n_base_points[i])
+        expected = to_vector(expected)
+        self.assertAllClose(result, expected)
+
+        n_tangent_vecs = self.space.projection_to_tangent_space(
+            n_vecs, base_point=n_base_points)
+        result = self.metric.exp(n_tangent_vecs, n_base_points)
+        self.assertVector(result, n_samples, dim)
+
+        expected = gs.zeros((n_samples, dim))
+        for i in range(n_samples):
+            expected[i] = self.metric.exp(n_tangent_vecs[i], n_base_points[i])
+        expected = to_vector(expected)
+        self.assertAllClose(result, expected)
+
+    def check_shape_log_vectorization(self, space, n_samples, dim):
+        one_point = self.space.random_uniform()
+        one_base_point = self.space.random_uniform()
+        n_points = self.space.random_uniform(n_samples=n_samples)
+        n_base_points = self.space.random_uniform(n_samples=n_samples)
+
+        result = self.metric.log(one_point, one_base_point)
+        self.assertVector(result, 1, dim)
+
+        result = self.metric.log(n_points, one_base_point)
+        self.assertVector(result, n_samples, dim)
+
+        result = self.metric.log(one_point, n_base_points)
+        self.assertVector(result, n_samples, dim)
+
+        result = self.metric.log(n_points, n_base_points)
+        self.assertVector(result, n_samples, dim)
+
+    def check_shape_squared_dist_vectorization(self, space, metric, n_samples):
+        one_point_a = self.space.random_uniform()
+        one_point_b = self.space.random_uniform()
+        n_points_a = self.space.random_uniform(n_samples=n_samples)
+        n_points_b = self.space.random_uniform(n_samples=n_samples)
+
+        result = self.metric.squared_dist(one_point_a, one_point_b)
+        self.assertScalar(result, 1)
+
+        result = self.metric.squared_dist(n_points_a, one_point_b)
+        self.assertScalar(result, n_samples)
+
+        result = self.metric.squared_dist(one_point_a, n_points_b)
+        self.assertScalar(result, n_samples)
+
+        result = self.metric.squared_dist(n_points_a, n_points_b)
+        self.assertScalar(result, n_samples)
+
+    def check_shape_dist_vectorization(self, space, metric, n_samples):
+        one_point_a = self.space.random_uniform()
+        one_point_b = self.space.random_uniform()
+        n_points_a = self.space.random_uniform(n_samples=n_samples)
+        n_points_b = self.space.random_uniform(n_samples=n_samples)
+
+        result = self.metric.dist(one_point_a, one_point_b)
+        self.assertScalar(result, 1)
+
+        result = self.metric.dist(n_points_a, one_point_b)
+        self.assertScalar(result, n_samples)
+
+        result = self.metric.dist(one_point_a, n_points_b)
+        self.assertScalar(result, n_samples)
+
+        result = self.metric.dist(n_points_a, n_points_b)
+        self.assertScalar(result, n_samples)
+
+
+def to_scalar(expected):
```

Comment:
the logic in this function is weird - you say to_scalar and you make this a rank 2 tensor

Reply:
Yes, it should be: to_our_convention_for_scalar, which is a (1, 1) tensor.